sdk/core/System.ClientModel/src/Options/PipelineMessageClassifier.cs

```diff
@@ -1,25 +1,91 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.IO;
+
 namespace System.ClientModel.Primitives;
 
 public class PipelineMessageClassifier
 {
-    internal static PipelineMessageClassifier Default { get; } = new PipelineMessageClassifier();
+    public static PipelineMessageClassifier Default { get; } = new EndOfChainClassifier();
 
     public static PipelineMessageClassifier Create(ReadOnlySpan<ushort> successStatusCodes)
         => new ResponseStatusClassifier(successStatusCodes);
 
-    protected internal PipelineMessageClassifier() { }
+    protected PipelineMessageClassifier() { }
+
+    public virtual bool TryClassify(PipelineMessage message, out bool isError)
+    {
+        isError = false;
+        return false;
+    }
+
+    public virtual bool TryClassify(PipelineMessage message, Exception? exception, out bool isRetriable)
+    {
+        isRetriable = false;
+        return false;
```

Comment:
Default implementation in base type doesn't classify so that inheritors don't accidentally stop the composition chain.  With this approach, derived types only need to implement classification routines where they do want to classify.  Default "End of Chain" functionality is provided as an immutable static instance.

Reply:
Why don't we make these abstract then?

Reply:
Subtypes would have to override both, even if they don't have an opinion about one type of classification.  Is that preferable?

Reply:
I think it would be better. Such override would be super simple (return false) and abstract members would make the API very transparent. As of now, I need to read docs to figure out what the default behavior is.

Reply:
SG.  I'll update it.